python/lsst/daf/butler/registry/queries/expressions/_predicate.py

```diff
@@ -78,6 +79,7 @@ def make_string_expression_predicate(
     string: str,
     dimensions: DimensionGraph,
     *,
+    column_types: ColumnTypeInfo,
```

Comment:
It's a bit of a shame that we have to pass `column_types` here (and through all the classes in `_structs.py`), since `QueryBuilder` already has access to that and that's enough in all other cases.  But I don't see a way out of it, at least not until some distant-future date when we've gotten rid of all `datetime` usage.